packages/graph/hash_graph/lib/graph/src/types/schema/mod.rs

```diff
@@ -0,0 +1,45 @@
+mod data;
+mod helper;
+pub mod properties;
+mod reference;
+
+use core::fmt;
+
+pub trait Validate {
+    /// Validates the schema of the object.
+    ///
+    /// This does only check the validity of this object, not it's child data. A validation also
+    /// happens when the type is created.
+    ///
+    /// # Errors
+    ///
+    /// - [`ValidationError`] if the object is not in a valid state.
+    fn validate(&self) -> Result<(), ValidationError>;
+}
+
+#[doc(inline)]
+pub use self::{
+    data::DataType,
+    helper::{Array, OneOf, OneOrMany},
+    properties::PropertyType,
+    reference::{DataTypeReference, PropertyTypeReference, Uri},
+};
+
+#[derive(Debug)]
+pub enum ValidationError {
+    PropertyMissing(Uri),
+    OneOfEmpty,
+}
+
+impl fmt::Display for ValidationError {
+    fn fmt(&self, fmt: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::PropertyMissing(uri) => {
+                write!(fmt, "Required field {uri} is not used in `properties`")
```

Comment:
Perhaps something like "Schema has marked the {uri} property as required, but it wasn't defined in the "properties" object"

Or something more clear about "property {uri} in `required` array was missing in `properties` object"

Reply:
Applied in 2613db2c7d1ffa73acc2061c21a5c0a8aa4f36c3